embassy-nrf/src/gpiote.rs

```diff
@@ -29,17 +29,14 @@ const CHANNEL_COUNT: usize = 12;
 /// Max channels per port
 const CHANNELS_PER_PORT: usize = 8;
 
-#[cfg(any(
-    feature = "nrf52833",
-    feature = "nrf52840",
-    feature = "_nrf5340",
-    feature = "_nrf54l15",
-    feature = "_nrf54l10",
-    feature = "_nrf54l05"
-))]
+#[cfg(any(feature = "nrf52833", feature = "nrf52840", feature = "_nrf5340"))]
 const PIN_COUNT: usize = 48;
+#[cfg(any(feature = "_nrf54l15", feature = "_nrf54l10", feature = "_nrf54l05"))]
+// Highest pin is P2.10 => (2 * 32 + 10) = 74
+const PIN_COUNT: usize = 74;
```

Comment:
maybe it needs a +1? if the array has length `PIN_COUNT` the valid indices are `0..PIN_COUNT-1`, and 74 here is the highest valid index.